src/nrtk_explorer/library/dataset.py

```diff
@@ -1,60 +1,213 @@
+        pass
 
-class DefaultDataset:
-    """Default dataset class to load the dataset and get the image file path given an image id."""
+    def get_image(self, id: int):
+        """Get the image given an image id."""
+        img = self._get_image(id)
+        # transforms and base64 encoding require RGB mode
+        return img.convert("RGB") if img.mode != "RGB" else img
+
+
+class JsonDataset(BaseDataset):
+    """JSON-based COCO datasets."""
 
     def __init__(self, path: str):
         with open(path) as f:
             self.data = json.load(f)
-            self.fpath = path
-            self.cats = {cat["id"]: cat for cat in self.data["categories"]}
-            self.anns = {ann["id"]: ann for ann in self.data["annotations"]}
-            self.imgs = {img["id"]: img for img in self.data["images"]}
+        self.fpath = path
+        self.cats = {cat["id"]: cat for cat in self.data["categories"]}
+        self.anns = {ann["id"]: ann for ann in self.data["annotations"]}
+        self.imgs = {img["id"]: img for img in self.data["images"]}
 
-    def get_image_fpath(self, selected_id: int):
-        """Get the image file path given an image id."""
+    def _get_image_fpath(self, selected_id: int):
         dataset_dir = Path(self.fpath).parent
         file_name = self.imgs[selected_id]["file_name"]
         return str(dataset_dir / file_name)
 
+    def _get_image(self, id: int):
+        image_fpath = self._get_image_fpath(id)
+        return Image.open(image_fpath)
 
-@lru_cache
-def __load_dataset(path: str):
-    """Load the dataset given the path to the dataset file."""
+
+def make_coco_dataset(path: str):
     try:
         import kwcoco
 
-        return kwcoco.CocoDataset(path)
+        class CocoDataset(kwcoco.CocoDataset, BaseDataset):
+            def _get_image(self, id: int):
+                image_fpath = self.get_image_fpath(id)
+                return Image.open(image_fpath)
+
+        return CocoDataset(path)
     except ImportError:
-        return DefaultDataset(path)
+        return JsonDataset(path)
+
+
+def is_coco_dataset(path: str):
+    if not os.path.exists(path):
+        return False
+    required_keys = ['"images"', '"categories"', '"annotations"']
+    with open(path) as f:
+        content = f.read()
+    return all(key in content for key in required_keys)
+
+
+def expand_hugging_face_datasets(dataset_identifiers: SequenceType[str], streaming=True):
+    expanded_identifiers = []
+    for identifier in dataset_identifiers:
+        if is_coco_dataset(identifier):
+            expanded_identifiers.append(identifier)
+        else:
+            infos = get_dataset_infos(identifier)
+            for config_name, info in infos.items():
+                for split_name in info.splits:
+                    streaming_str = "streaming" if streaming else "download"
+                    expanded_identifiers.append(
+                        f"{identifier}@{config_name}@{split_name}@{streaming_str}"
```

Comment:
This is bananas 🍌  Is there a better way\ to bring some "config" along with the hugging face dataset?